docs/design/RUC-on-type.md

```diff
@@ -0,0 +1,189 @@
+
+Also, RequiresUnreferencedCode on type does not suppress the warnings inside a nested type in any way.
+
+```C#
+[RequiresUnreferencedCode ("Message for --ClassWithRequiresUnreferencedCode--")]
+class ClassWithRequiresUnreferencedCode
+{
+    class NestedClass {
+        // ILXXXX: The method will generate a warning that will not be suppressed
+        public static void MethodWithDangerousCode ()
+        { 
+            // Dangerous code that generetares a trimming warning
+        }
+    }
+}
+
+static void TestSuppressionOnType ()
+{
+    ClassWithRequiresUnreferencedCode.NestedClass.MethodWithDangerousCode()
+}
+```
+
+## Derived types
+There is a special handling for types deriving from a base type which is annotated with RequiresUnreferencedCode. Although we dont auto propagate the RequiresUnreferencedCode attribute to the derived type a warning is generated in the derived type declaration to warn the user that calling methods in the derived type might end up calling something dangerous at it's base. Therefore, is recomendable to also mark the derived type with RequiresUnreferencedCode.
+
+```C#
+[RequiresUnreferencedCode ("Message for --ClassWithRequiresUnreferencedCode--")]
+class ClassWithRequiresUnreferencedCode
+{
+    public static void StaticMethod () { }
+}
+
+// IL2109: Type 'DerivedWithoutRequires' derives from 'ClassWithRequiresUnreferencedCode' which has 'RequiresUnreferencedCodeAttribute'.  Message for --ClassWithRequiresUnreferencedCode--.
+class DerivedWithoutRequires : ClassWithRequiresUnreferencedCode
+{
+    public static void StaticMethodInInheritedClass () { }
+}
+
+// TestRequiresOnBaseButNotOnDerived(): Using method 'ClassWithRequiresUnreferencedCode.StaticMethod()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Message for --ClassWithRequiresUnreferencedCode--.
+static void TestRequiresOnBaseButNotOnDerived ()
+{
+    DerivedWithoutRequires.StaticMethodInInheritedClass ();
+    DerivedWithoutRequires.StaticMethod ();
+}
+```
+As the previous example ilustrates the DerivedType without RequiresUnreferencedCode will generate a warning only for deriving from ClassWithRequiresUnreferencedCode. Additionally, by calling some of the methods we can see the difference on behavior on methods declared on the derived type vs methods on the base type. Calling a method in the derived type will not generate a warning since the derived type is not annotated, while calling a method for which the base type will be called will generate IL2026.
+
+Arguably we could say that the warning in the type declaration is not necessary and that we could rely on the warning being generated in the constructor instead, but this could lead to some behaviors in which we can call a method without ever using the constructor.
```

Comment:
The argument was that there's nothing wrong about not getting a warning in this case. The Derived.SomeMethod doesn't have any dangerous code in it - if it did it would get a warning of its own.

Assuming we can "trust" in us detecting all calls to base (specifically that `.ctor` on Derived will always call `.ctor` on Base), then I don't see a need for a warning on inheritance.

Reply:
Deleted the scenario and the documentation about warning with IL2109